src/org/sosy_lab/java_smt/solvers/bitwuzla/BitwuzlaFormulaManager.java

```diff
@@ -137,12 +137,16 @@ public Term parseImpl(String formulaStr) throws IllegalArgumentException {
     // Process the symbols from the parser
     Map_TermTerm subst = new Map_TermTerm();
     for (Term term : declared) {
-      if (cache.containsRow(term.symbol())) {
+      String symbol = term.symbol();
+      if (symbol.startsWith("|") && symbol.endsWith("|")) {
```

Comment:
I find it dangerous to perform this operation manually with inline string manipulation. Isn't there more that needs to be done? For example unescaping or further symbols, such as nested `|`? And obviously that should be a general utility function somewhere where it is properly maintainable, with unit tests, links to the standard, etc. instead of hidden deeply inside some other code where nobody will be able to find and update it if one for example needs to update the escaping.

Reply:
Well its OK if its thoroughly documented and tested in my opinion. The handling is unique for this solver and best-effort based and can be improved at any point.

But we definitely need tests for this and similar cases. (Multiple `|`, other special characters in names etc.)

Reply:
I've refactored the code and unescaping for SMTLIB variables is now handled directly in the variable cache itself. This should avoid similar bugs where we forget to unescape symbols when accessing the cache.

>  Isn't there more that needs to be done? For example unescaping or further symbols, such as nested |? 

This is some very low level code and we will only encounter variable names that are valid in SMTLIB. This means there can't be any nested `|` and the `escape`/`unescape` functions from `FormulaManager` should also be unaffected.

Reply:
If this issue only appears when parsing SMTLIB, why do we need to handle quoting in the complete cache and not only in the parser?

Reply:
> If this issue only appears when parsing SMTLIB, why do we need to handle quoting in the complete cache and not only in the parser?

Hello Karlheinz,
I've refactored the code once more and restored the formula cache to its original version. Removing SMTLIB quotes is now handled in a helper function named `BitwuzlaCreator.removeQuotes`, which needs to be called from `BitwuzlaFormulaManager.parseImpl` and `BitwuzlaCreator.visit`. I missed `BitwuzlaCreator.visit` in my original patch, but noted that several other crashes went away when I removed the quotes for the entire cache. The new version of the code should  now cover both cases without requiring any changes to the cache itself.